src/world/partner/watt.c

```diff
@@ -1,29 +1,169 @@
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BD710_31D280);
+void func_802BD710_31D280(Npc* npc) {
+    if (D_8010C954 != 0) {
+        D_8010C954 = 0;
+        npc->flags = D_802BE274_31DDE4->unk_08;
+        D_802BE274_31DDE4->unk_04 = 0;
+        partner_clear_player_tracking(npc);
+    }
+}
 
 INCLUDE_ASM(s32, "world/partner/watt", func_802BD754_31D2C4);
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BDD0C_31D87C);
+void force_player_anim(s32 arg0);
+s32 func_802BDD0C_31D87C(Evt* script, s32 isInitialCall) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    Npc* npc = script->owner2.npc;
+    
+    if (isInitialCall != 0) {
+        func_802BD180_31CCF0();
+        partner_init_put_away(npc);
+        force_player_anim(0x10002);
+        partnerActionStatus->actionState.b[3] = 0;
+        partnerActionStatus->actionState.b[0] = 0;
+        playerStatus->animFlags = playerStatus->animFlags & ~3;
+        gGameStatusPtr->unk_7D = 0;
+    }
+    return partner_put_away(npc) != FALSE;
+}
 
 INCLUDE_ASM(s32, "world/partner/watt", func_802BDD9C_31D90C);
 
 INCLUDE_ASM(s32, "world/partner/watt", func_802BDE10_31D980);
 
-INCLUDE_ASM(s32, "world/partner/watt", func_802BDE88_31D9F8);
+void func_802BE070_31DBE0(void);
+void partner_kill_ability_script(void);
+s32 partner_use_ability(void);
+ApiStatus func_802BDE88_31D9F8(Evt* script, s32 isInitialCall) {
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Npc* npc;
+
+    npc = get_npc_unsafe(-4);
+    if (isInitialCall != 0) {
+        script->functionTemp[0] = 0;
+    }
+
+    switch (script->functionTemp[0]) {
+        case 0:
+            if (script->varTable[12] == 0) {
+                partner_kill_ability_script();
+                func_802BD180_31CCF0();
+            } else {
+                func_802BD100_31CC70(npc);
+            }
+
+            script->functionTemp[1] = script->varTable[4];
+            playerStatus->targetYaw = atan2(playerStatus->position.x, playerStatus->position.z, script->varTable[1], script->varTable[3]);
+            playerStatus->heading = playerStatus->targetYaw;
+            move_player(script->functionTemp[1], playerStatus->heading, *((f32*) &script->varTable[5]));
+            func_802BE070_31DBE0();
+            npc->flags = (npc->flags & ~0x200) | 0x100;
+            playerStatus->animFlags = playerStatus->animFlags | 3;
+            gGameStatusPtr->unk_7D = 1;
+            partnerActionStatus->actionState.b[0] = 1;
+            partnerActionStatus->actionState.b[3] = 6;
+            D_802BE308 = 0;
+            script->functionTemp[0] += 1;
+            break;
+        case 1:
+            func_802BE070_31DBE0();
+            script->functionTemp[1]--;
+            if (script->functionTemp[1] == 0) {
+                if (script->varTable[12] != 0) {
```

Comment:
It's being treated like a bool
```suggestion
                if (script->varTable[12]) {
```